litestar/dto/base_dto.py

```diff
@@ -50,6 +50,11 @@ class AbstractDTO(Generic[T]):
 
     _dto_backends: ClassVar[dict[str, _BackendDict]] = {}
 
+    def __init_subclass__(cls, **kwargs: Any) -> None:
+        super().__init_subclass__(**kwargs)
+        if hasattr(cls, "config"):
+            cls.config = cls.adapt_dto_config(cls.config)
+
```

Comment:
I don't like that this implementation detail of the `SQLAlchemyDTOConfig` bleeds over into this.

Maybe it could be handled solely in there?

Reply:
I was thinking more broadly that other factories would have specialized configs soon, this is also the reason why the `adapt` method was made public and run in the base dto init subclass.

Reply:
I'd prefer not committing to a public API for this as long as we don't have other use cases. Should those arise we can still make this API public.